rust/signed_doc/src/metadata/doc_type.rs

```diff
@@ -250,10 +290,64 @@ impl Encode<ProblemReport> for DocType {
     }
 }
 
+impl<'de> Deserialize<'de> for DocType {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where D: Deserializer<'de> {
+        #[derive(Deserialize)]
+        #[serde(untagged)]
+        enum DocTypeInput {
+            /// Single UUID string.
+            Single(String),
+            /// List of UUID string.
+            Multiple(Vec<String>),
+        }
+
+        let input = DocTypeInput::deserialize(deserializer)?;
+        let dt = match input {
+            DocTypeInput::Single(s) => {
+                let uuid = Uuid::parse_str(&s).map_err(|_| {
+                    serde::de::Error::custom(DocTypeError::StringConversion(s.clone()))
+                })?;
+                // If there is a map from old (single uuid) to new use that list, else convert that
+                // single uuid to [uuid] - of type DocType
+                map_doc_type(uuid).unwrap_or(uuid.try_into().map_err(serde::de::Error::custom)?)
+            },
+            DocTypeInput::Multiple(v) => v.try_into().map_err(serde::de::Error::custom)?,
+        };
+        Ok(dt)
+    }
+}
+
+impl PartialEq for DocType {
```

Comment:
Pls add a comment that this needs to preserve backward compatibility with the old solution